backend/utils.py

```
@@ -77,13 +82,16 @@ def _validate_uploaded_file_sync(file: UploadFile) -> Optional[Image.Image]:
+            detected_mime = magic.from_buffer(file_content, mime=True)
 
-        if detected_mime not in ALLOWED_MIME_TYPES:
-            raise HTTPException(
-                status_code=400,
-                detail=f"Invalid file type. Only image files are allowed. Detected: {detected_mime}"
-            )
+            if detected_mime not in ALLOWED_MIME_TYPES:
+                raise HTTPException(
+                    status_code=400,
+                    detail=f"Invalid file type. Only image files are allowed. Detected: {detected_mime}"
+                )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)